README.md

```diff
@@ -1 +1,588 @@
+function scheduleUpdate() {
+  if(!updateRequested) {
+    updateRequested = true;
+    updateRequested = await false;
+    update();
+  }
+}
+```
+
+Since, from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await):
+> If the value of the expression following the await operator is not a Promise, it's converted to a resolved Promise.
+
+In fact, [Justin Fagnani](https://twitter.com/justinfagnani), one of the authors of LitElement, once told me on twitter that the pre-LitElement version did exactly this:
+
+![twitter screenshot](./img/justin.png)
+
+### Deferring promises
+
+Alright, we're almost getting to the good stuff now. But before we go there, there's one final concept we need to explore, and thats deferring resolving of Promises, which will provide us with a handy utility later on in this blog post.
+
+Take a look at the following code:
+
+```js
+/* We declare a new variable */
+let resolveFn;
+
+/** 
+ * We declare a new promise, but we _dont_ resolve it! 
+ * Instead, we assign the `resolve` function to the
+ * `resolveFn` that we declared above
+ */
+const myPromise = new Promise((resolve) => {
+  resolveFn = resolve;
+});
+
+/** 
+ * `myPromise` now is just a pending promise, waiting to be resolved,
+ * and we can do that whenever we please with our `resolveFn`
+ */
+myPromise; // Promise {<pending>}
+
+/* For example, we could do a whole bunch of other work in between */
+
+/* And finally, whenever we're ready, we can call the `resolveFn` to resolve `myPromise` */
+resolveFn(); // Resolves `myPromise`
+
+/* Aaand `myPromise` is now fulfilled! */
+myPromise; // Promise {<fulfilled>: undefined} ! 🎉
+```
+
+This pattern is often used in cases (like unit tests) where you have to await a specific amount of time before continuing.
+
+```js
+function sleep(delay) {
+  return new Promise(resolve => {
+    setTimeout(resolve, delay)
+  });
+}
+
+(async () => {
+  await sleep(1000); // Awaits one second for the timeout to execute before continuing
+})()
+```
+
+In this example, the `sleep` function returns a new `Promise`. Inside the promise, we set a timeout, but we only resolve the promise once the `setTimeout` callback is executed. So by `await`ing the `sleep` function, we essentially force to wait until `setTimeout` callback has executed.
+
+Be aware, however, that JavaScript is single-threaded, and blocking the main thread is generally considered a bad practice.
+
+## Back to LitElement
+
+### 🧙‍♂️ Automagically requested updates
+
+LitElement is _reactive_. What this means is that, like many other modern frontend libraries, we dont have to worry as much about manually rendering or re-rendering the contents of our component. All we have to do as a user of LitElement is declare our properties, LitElement will then _observe_ these properties for us, and when a property has changed LitElement takes care of updating our component for us. Nice. I like not doing work.
+
+![counter component](./img/wcdev.png)
```

Comment:
<plug>
If you do this, than any page with the element registered (like ours, when we run the build we'll detect that the element is used and append) will get a live preview, but everyone else will get the image. 
</plug>

```suggestion
<wcd-live title="Counter Component">

[![counter component](./img/wcdev.png)](https://webcomponents.dev/edit/THE_ID)

</wcd-live>
```

Reply:
How would this work on medium? 🤔

Reply:
since medium doesn't support markdown, whatever tool you're using would just strip out the tags

https://markdowntomedium.com/